internal/approver/evaluator/evaluator.go

```diff
@@ -98,16 +96,12 @@ func (i *internal) Evaluate(req *cmapi.CertificateRequest) error {
 			csr.DNSNames, csr.IPAddresses, csr.Subject.CommonName, csr.EmailAddresses)
 	}
 
-	if ecdsapub, ok := csr.PublicKey.(*ecdsa.PublicKey); !ok || ecdsapub.Curve.Params().BitSize != 521 {
-		return errors.New("forbidden key used by requestor, expecting ECDSA P521")
-	}
-
```

Comment:
We use the same verification in cert-manager, so that presumably would be incorrect too https://github.com/cert-manager/cert-manager/blob/b5d1b1e27467e7fd9be4124f0041ec58524105c4/pkg/util/pki/match.go#L91-L93. I had a brief look as to where this could possibly come from by looking at Kuberentes codebase and imports and found that it looks like a similar check is used in go jwt library [here](https://github.com/golang-jwt/jwt/blob/v4.5.0/ecdsa.go#L120-L124) (`m.CurveBits` would be set to 256 [here](https://github.com/golang-jwt/jwt/blob/v4.5.0/ecdsa.go#L34)). Do you think this check is also invalid?
If so is there another way we can check that the curve size is 256? It seems like that if this is one of the things we promise to the users will be true about the certificate request (alongside it being for the particular SPIFFE identity) then the approver should verify that?